dfusion_rust_core/src/models/solution.rs

```diff
@@ -17,12 +16,65 @@ pub struct Solution {
 impl Solution {
     pub fn trivial(num_orders: usize) -> Self {
         Solution {
-            objective_value: Some(U256::zero()),
             prices: vec![0; TOKENS as usize],
             executed_buy_amounts: vec![0; num_orders],
             executed_sell_amounts: vec![0; num_orders],
         }
     }
+
+    /// Returns the token price for token or `None` if the token is invalid.
+    fn price(&self, token: u16) -> Option<u128> {
+        self.prices.get(token as usize).cloned()
+    }
+
+    /// Compute the result of the objective function for the solution. This is
+    /// currently defined as *U - dUT* where *U* is the total utility of the
+    /// trade and *dUT* is the total disreguarded utility of touched orders.
+    ///
+    /// # Returns
+    ///
+    /// Returns `Some(U256)` for any solution that is positive valid solution.
+    /// Returns `None` for invalid solutions or orders. A solution might be
+    /// invalid if:
+    /// - The number of orders do not match the solution buy/sell amounts
+    /// - The solution values lead to invalid utility or disregarded utility
+    ///   results
+    /// - The orders reference tokens that are not priced by the solution
+    pub fn objective_value(&self, orders: &[Order]) -> Option<U256> {
+        if orders.len() != self.executed_buy_amounts.len()
+            || orders.len() != self.executed_sell_amounts.len()
```

Comment:
There will likely be many more orders in an auction than can be touched. Is this condition prepared to handle that? Are we assuming that `objective_value` is only ever being passed a collection of touched orders?

Reply:
From what I understand the solution `executed_buy_amounts` and `executed_sell_amounts` are created with the length of `num_orders` and are just 0 for untouched orders.

Reply:
If this isn't true anymore, then we would have to change the solution function so that `excuted_{buy,sell}_amount` is a `map(order_id -> u128)` instead of just a vector; or else I don't see a way to recover the values for a given order.